src/rosdistro/rosdistro.py

```diff
@@ -199,24 +199,24 @@ def _fetch_package_xml(self, rosdistro):
         repo = self.repository
         if 'github.com' in repo.url:
             url = repo.url
-            upstream_version = repo.version.split('-')[0]
-            release_tag = 'release/{0}/{1}'.format(self.name, upstream_version)
-            url = url.replace('.git', '/{0}/package.xml'.format(release_tag))
+            # release_tag = get_release_tag(self.name)
+            tail = '/{0}/package.xml'
+            url = url.replace('.git', tail)
             url = url.replace('git://', 'https://')
             url = url.replace('https://', 'https://raw.')
+
             try:
                 try:
+                    release_tag = 'release/{0}/{1}/{2}'.format(rosdistro, self.name, repo.version)
+                    url = url.format(release_tag)
```

Comment:
Before `format` was only called on strings defined in the source code. Now `format` is being called on a user provided string which might contain characters like `{}` and therefore result in problems. Therefore please only use `format` on strings defined in the code directly.

Same below in line 218:

> url = url.format(release_tag)